native/src/main/java/io/ballerina/stdlib/http/api/HttpCallableUnitCallback.java

```diff
@@ -73,17 +81,59 @@ public void notifySuccess(Object result) {
             invokeErrorInterceptors((BError) result, true);
             return;
         }
-        returnResponse(result);
+        returnResponse(result, checkAddingLinks(result));
     }
 
-    private void returnResponse(Object result) {
-        Object[] paramFeed = new Object[6];
+    private boolean checkAddingLinks(Object result) {
+        if (links != null && !links.isEmpty()) {
+            if (result instanceof BMap && !((BMap) result).containsKey(LINKS)) {
+                return addLinksToMap((BMap) result);
+            }
+        }
+        return false;
+    }
+
+    private boolean addLinksToMap(BMap result) {
```

Comment:
What is the reason to return a `boolean` from this method ?

Reply:
The `boolean` represents if the method succeed or not? 
In this case the method name should be `checkAddingLinksToMap`?

Reply:
In Java we use checked exceptions to indicate the failure of a method execution (even though we have some exceptions in Java API itself). Following is a good read about using `boolean` values to indicated success or failure.
[https://stackoverflow.com/questions/15347034/what-is-a-good-practise-of-executing-some-operation-and-returning-boolean-value](https://stackoverflow.com/questions/15347034/what-is-a-good-practise-of-executing-some-operation-and-returning-boolean-value)

Reply:
@ayeshLK Thanks, will try to refactor the code 👍

Reply:
Refactored the code in [c08fcec](https://github.com/ballerina-platform/module-ballerina-http/pull/1068/commits/c08fcecc7d972b5883ac1eddc4e946b95e8a9836)